File: chapter05/functions_2_3.clj
(def ->fizzbuzz
  (fn [rem3 rem5 n]
    (cond
      (and (zero? rem3) (zero? rem5)) "FizzBuzz"
      (zero? rem3) "Fizz"
      (zero? rem5) "Buzz"
      :else n)))

(def fizzbuzz
  (fn [n] (->fizzbuzz (rem n 3) (rem n 5) n)))

File: chapter05/functions_2_3.exs
to_fizzbuzz = fn
  (0, 0, _) -> "FizzBuzz"
  (0, _, _) -> "Fizz"
  (_, 0, _) -> "Buzz"
  (_, _, n) -> n
end

fizzbuzz = fn(n) -> to_fizzbuzz.(rem(n, 3), rem(n, 5), n) end
